### ISSUE TITLE
Agent goes unhealthy on assigning to agent policy not having Endpoint integration.

### ISSUE THREAD

**Issue by amolnater-qasource** [elastic/elastic-agent#1282]
**Kibana version:** 8.5 BC1 Kibana cloud environment

**Host OS:** Windows 10, Windows 2022

**Build details:**
VERSION: 8.5 BC1
BUILD: 56595
COMMIT: 0d8de4df69f8084a94cdd9638d7de510813cb5ce

**Preconditions:**
1. 8.5 BC1 Kibana cloud environment should be available.
2. Windows agent should be installed with Policy having System and Endpoint Defend integration.

**Steps to reproduce:**
1. Navigate to Fleet> Agents policies tab.
2. Create a new policy with only System integration.
3. Assign windows agent to this policy.
4. Observe agent goes unhealthy.

**Screenshot:**
![3](https://user-images.githubusercontent.com/77374876/191951817-677ce114-d3aa-4948-942f-5d35838520e1.JPG)

**Logs:**
[elastic-agent-diagnostics-2022-09-23T11-28-07Z-00.zip](https://github.com/elastic/elastic-agent/files/9633193/elastic-agent-diagnostics-2022-09-23T11-28-07Z-00.zip)

**Expected Result:**
Agent should remain Healthy on assigning to agent policy not having Endpoint integration.

**Comment by amolnater-qasource** [comment]
@manishgupta-qasource Please review.

**Comment by manishgupta-qasource** [comment]
Secondary review for this ticket is Done

**Comment by cmacknz** [comment]
Possibly related to https://github.com/elastic/elastic-agent/issues/1262 CC @aleksmaus

**Comment by cmacknz** [comment]
Interesting log fragments:

```json
{"log.level":"info","@timestamp":"2022-09-23T10:51:57.711Z","log.origin":{"file.name":"stateresolver/stateresolver.go","file.line":48},"message":"New State ID is x37iTOax","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-09-23T10:51:57.711Z","log.origin":{"file.name":"stateresolver/stateresolver.go","file.line":49},"message":"Converging state requires execution of 2 step(s)","ecs.version":"1.6.0"}
{"log.level":"error","@timestamp":"2022-09-23T10:51:57.711Z","log.origin":{"file.name":"emitter/controller.go","file.line":123},"message":"Failed to render configuration with latest context from composable controller: operator: failed to execute step sc-run, error: context canceled: context canceled","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-09-23T10:51:57.819Z","log.logger":"reexec","log.origin":{"file.name":"reexec/reexec_windows.go","file.line":36},"message":"Running as Windows service Elastic Agent; triggering service restart","ecs.version":"1.6.0"}
```

```json
{"log.level":"info","@timestamp":"2022-09-23T11:16:18.102Z","log.origin":{"file.name":"log/reporter.go","file.line":40},"message":"2022-09-23T11:16:18Z - message: Application: endpoint-security--8.5.0[694d58e5-bfe3-41a2-bc23-2932c4b1fb79]: State changed to STOPPING: Protecting with policy {cf42ec3c-1663-454d-8113-e81c22ab61fd} - type: 'STATE' - sub_type: 'STOPPING'","ecs.version":"1.6.0"}
{"log.level":"error","@timestamp":"2022-09-23T11:16:46.617Z","log.origin":{"file.name":"status/reporter.go","file.line":259},"message":"Elastic Agent status changed to \"error\": \"app endpoint-security--8.5.0-518dff79: failed to stop after 30s: application stopping timed out\"","ecs.version":"1.6.0"}
{"log.level":"error","@timestamp":"2022-09-23T11:16:46.617Z","log.origin":{"file.name":"log/reporter.go","file.line":36},"message":"2022-09-23T11:16:46Z - message: Application: endpoint-security--8.5.0[694d58e5-bfe3-41a2-bc23-2932c4b1fb79]: State changed to FAILED: failed to stop after 30s: application stopping timed out - type: 'ERROR' - sub_type: 'FAILED'","ecs.version":"1.6.0"}
```

**Comment by cmacknz** [comment]
This might be https://github.com/elastic/endpoint/issues/29

**Comment by jlind23** [comment]
@amolnater-qasource can we test this again with the patch coming from endpoint team which should now be available on 8.5 BC + Snapshot.

**Comment by simranvaseer-qasource** [comment]
### `Bug Conversion`

-  **01 new testcase** is created under the 'Fleet Test Suite' at:
   - https://elastic.testrail.io/index.php?/cases/view/168354

Thanks!
